### DISCUSSION TITLE
How do you modify docker container configs?

### DISCUSSION THREAD

**Discussion by jessepage1989** [Security-Onion-Solutions/securityonion#8556] [Unsupported Versions]
I am currently working through getting SO to be accessible via a reverse proxy. I am trying to edit the kratos settings. When going through readthedocs, information about kratos is pretty much a reference to kratos' docs. `SOC auth is handled by Kratos and you can read more about that at [https://github.com/ory/](https://github.com/ory/kratos)` 
Kratos settings are configured via a yaml config file. It appears the kratos container's config file is located at `/opt/so/conf/kratos/kratos.yaml`
When I edit this file however, the settings are wiped at restart when I run `docker container restart so-kratos`
How do I change the kratos config? 

Also if different, how would you modify the nginx config?

**Comment by xfaith** [comment]
I would look at the global.sls or the minions.sls files and look for matching kratos information in there.   If you want to provide more details I might be able to do some more digging.  As a reminder salt is what is basically building the docker containers.  You could also do sort of a reverse lookup and do a docker inspect on the kratos container and figure out the bindings, and find that those files (once again its most likely in the /opt/so/salt/default (move it to local)

**Reply by jessepage1989** [reply]
The bindings from running `docker inspect so-kratos` reference the files at /opt/so/conf/kratos/*  I don't see anything mentioning kratos in the global.sls file or the minion file. The only thing I found in the read the docs is this https://docs.securityonion.net/en/2.3/soc-customization.html?highlight=kratos#session-timeout which mentions editing the minion file to change the session timeout value. I'm not sure how this translates to rewriting the kratos.yaml file as the naming convention used in the minion yaml is not the same as stated in the kratos.yaml. 
minion.sls = 
`kratos:
  kratoskey: '<String>'`
kratos.yaml = 
`secrets:
  default:
    - <String>`

I am trying to change the session cookie settings. Referenced here https://www.ory.sh/docs/kratos/guides/configuring-cookies

Currently when navigating to my SO instance in the browser I get a cookie error at `https://so.mydomain.net/auth/self-service/login/browser` 
I am trying to figure out how to change the "Strict Cookie enforcement" settings that SO mentions at the initial setup. 

I have a nginx reverse proxy (letsencrypt cert) that I use to access some self hosted services at `"<servicename>.mydomain.net"`. I am trying to get my SO instance to work with it for some remote lab practice with some colleagues.

**Reply by xfaith** [reply]
You might have to wait for SecOn team for more details.    

But  if you are editing the kratos.yaml  you could try  this:
cp /opt/so/saltstack/default/salt/kratos/files/kratos.yaml /opt/so/saltstack/local/salt/kratos/files/

copy the default kratos.yaml to local.  

I am unsure of what changes you are actually trying to test it.  I am using a VM locally so not the same setup.   

Hope that helps a bit.
